core/src/main/java/org/elasticsearch/rest/action/cat/RestTable.java

```diff
@@ -109,6 +114,31 @@ public static RestResponse buildTextPlainResponse(Table table, RestChannel chann
         return new BytesRestResponse(RestStatus.OK, BytesRestResponse.TEXT_CONTENT_TYPE, bytesOut.bytes());
     }
 
+    static List<Integer> getRowOrder(Table table, RestRequest request) {
+        String[] columnOrdering = request.paramAsStringArray("s", null);
+
+        List<Integer> rowOrder = new ArrayList<>();
+        for(int i = 0; i < table.getRows().size(); i++) {
+            rowOrder.add(i);
+        }
+
+        if(columnOrdering != null) {
+            Map<String, String> headerAliasMap = table.getAliasMap();
+            List<ColumnOrderElement> ordering = new ArrayList<>();
+            for(int i = 0; i < columnOrdering.length; i++) {
+                String[] orderingEntry = columnOrdering[i].split(":");
+                boolean reverse = (orderingEntry.length > 1 && orderingEntry[1].equals("desc"));
```

Comment:
I'd do something like
```
boolean reverse = false;
if (columnOrdering[i].endsWith(":desc")) {
  columnOrder[i] = columnOrder[i].substring(...);
} else if (columnOrdering[i].endsWith(":asc")) {
  columnOrder[i] = columnOrder[i].substring(...);
}
```

or something like that. I think we should complain if it ends in something other than `:desc` and `:asc`.